### ISSUE TITLE
SamsungVR does not sync with phone correctly.

### ISSUE THREAD

**Issue by rasufe** [FredTungsten/ScriptPlayer#101]
Hello,

I used to have no problems using SamsungVR with my Galaxy s7 Edge, the entire setup was working wonderfully.

Now, all of the sudden it is no longer working. Seemingly nothing changed in between the time of it working and it not working. I can't figure this out! 

When I enable Presentation mode, I get the SamsungVR connected dialogue on ScriptPlayer. However, it seems to be only a 'fake' connection. If I play a video, it does not detect that a video is playing like it did before. The time is always set to 5 seconds / 00:00, and no matter what video I play on the S7 Edge (in Samsung VR Videos), it does not pick it up. It connects to ScriptPlayer just fine, but it won't play videos. 

Things I have tried:
1. Restarting all devices, same problem.
2. Changing the device from my laptop to my desktop, same problem.
3. Factory reset the s7 edge and start over, same problem.
4. Restored a nandroid backup from when I knew it was working, same problem.
5. Changed wifi settings from WPA2-PSK [AES] to WPA-PSK [TKIP] + WPA2-PSK [AES] (There is no TKIP only wifi setting on my router), same problem.
6. Now -- I switched to a brand new phone that was factory reset -- A Note 8. Set it all up and I get the same issue.
7. I changed the UDP ports randomly using user.cfg in milkVR (tried 4999 and 4500), no dice.
8. Turned off all firewalls on the network (even though it wasn't a problem before).
9. Tried with a second computer (laptop instead of my desktop), same problem.
10. Restarted the router.
11. Opened the port 5000 for the IP address just in case, but had no effect.

Still no solution. 

The issue seems to persist over devices (tried 2 phones, a laptop, and a desktop), and it came out of nowhere.  

Does anyone know how I can troubleshoot this? I don't know what kind of logs I can look at.

**Comment by rasufe** [comment]
I just noticed that the new Samsung VR Video App (2.09.9 from 4/30/19) has this in the patchnotes 'Deprecated support for Presentation Mode with Samsung VR Mobile' -- did this app update break the connection ability?

**Comment by rasufe** [comment]
Ugh.. I fixed it by uninstalling and rolling back to an older version. The only old version APK I could find was 2.04.2. See link below.
https://www.apkmirror.com/apk/samsung-electronics-co-ltd/samsung-vr-2/samsung-vr-2-2-04-2-release/samsung-vr-2-04-2-android-apk-download/

It appears that the new version 2.09.9 broke Presentation Mode.

**Comment by gagax1234** [comment]
can confirm version 2.09.9 breaks also on the oculus go :/
this is the only message that gets sent in presentation mode

```
{  
   "cmd":"txrx_broadcast_transmitter",
   "data":{  
      "ip":"192.168.1.171",
      "sendPort":5000,
      "broadcastPort":5001,
      "deviceName":"Oculus Go",
      "statusInterval":0
   }
}
```
I uninstalled the version I had and installed the one you posted and it worked again.
Thankfully the GearVR version is compatible with the oculus go.

**Comment by sboombastic3** [comment]
That apk won't install.  Anybody with other ideas?  Note9...

**Comment by rasufe** [comment]
It does install, you just need to uninstall the other APK first. May require going into the Galaxy or VR App store (I forgot which one) and uninstalling it that way.

**Comment by sboombastic3** [comment]
For some reason that kept giving me problems.  But I did find another with
that version and its working again.  I spent 3 hours talking to Samsung
asking why they did an update that didn't work.  I got one supervisor to
admit presentation mode works with no applications.  So i got farther up
the ladder and they didn't even know what presentation mode was...   Safe
to say we won't be seeing any patches....  supposedly, the higher up
manager I was speaking too would escalate to somebody else and I would hear
back.  Not holding my breath.  Then virtualrealporn stopped giving config
script files.  They only let you use feel connect and the script isn't
seperate.  So I'm screwed there too now.. At least at Wankzvr you get both
files.    ugh
